spark/src/main/java/io/delta/dynamodbcommitstore/integration_tests/dynamodb_commitowner_integration_test.py

```diff
@@ -0,0 +1,233 @@
+    .getOrCreate()
+
+# spark.sparkContext.setLogLevel("INFO")
+
+print("Creating table at path ", delta_table_path)
+spark.sql(f"CREATE table delta.`{delta_table_path}` (id int, a int) USING DELTA") # commit 0
+
+
+def write_tx(n):
+    print("writing:", [n, n])
+    spark.sql(f"INSERT INTO delta.`{delta_table_path}` VALUES ({n}, {n})")
+
+
+stop_reading = threading.Event()
+
+
+def read_data():
+    while not stop_reading.is_set():
+        print("Reading {:d} rows ...".format(
+            spark.read.format("delta").load(delta_table_path).distinct().count())
+        )
+        time.sleep(1)
+
+
+def start_read_thread():
+    thread = threading.Thread(target=read_data)
+    thread.start()
+    return thread
+
+
+print("===================== Starting reads and writes =====================")
+read_threads = [start_read_thread() for i in range(concurrent_readers)]
+pool = ThreadPool(concurrent_writers)
+start_t = time.time()
+pool.map(write_tx, range(num_rows))
+stop_reading.set()
+
+for thread in read_threads:
+    thread.join()
+
+print("===================== Evaluating number of written rows =====================")
+actual = spark.read.format("delta").load(delta_table_path).distinct().count()
+print("Actual number of written rows:", actual)
+print("Expected number of written rows:", num_rows)
+assert actual == num_rows
+
+t = time.time() - start_t
+print(f"{num_rows / t:.02f} tx / sec")
+
+current_table_version = num_rows
+dynamodb = boto3.resource('dynamodb', endpoint_url=dynamo_endpoint)
+ddb_table = dynamodb.Table(dynamo_table_name)
+
+def get_dynamo_db_table_entry_id():
+    table_properties = spark.sql(f"DESCRIBE DETAIL delta.`{delta_table_path}`").select("properties").collect()[0][0]
+    table_conf = table_properties.get(f"delta.managedCommits.tableConf{property_key_suffix}", None)
+    if table_conf is None:
+        return None
+    return json.loads(table_conf).get("tableId", None)
+
+def validate_table_version_as_per_dynamodb(expected_version):
+    table_id = get_dynamo_db_table_entry_id()
+    assert table_id is not None
+    print(f"Validating table version for tableId: {table_id}")
+    item = ddb_table.get_item(
+        Key={
+            'tableId': table_id
+        },
+        AttributesToGet = ['tableVersion']
+    )['Item']
+    current_table_version = int(item['tableVersion'])
+    assert current_table_version == expected_version
+
+delta_table_version = num_rows
+validate_table_version_as_per_dynamodb(delta_table_version)
+
+def perform_insert_and_validate(insert_value):
+    spark.sql(f"INSERT INTO delta.`{delta_table_path}` VALUES ({insert_value}, {insert_value})")
+    res = spark.sql(f"SELECT 1 FROM delta.`{delta_table_path}` WHERE id = {insert_value} AND a = {insert_value}").collect()
+    assert(len(res) == 1)
+
+def check_for_delta_file_existence(version, is_backfilled, should_exist):
```

Comment:
maybe call them `check_for_delta_file` because existence is decided by `should_exist` param.

Reply:
Renamed to `check_for_delta_file_in_filesystem` to make it explicit that we are querying the filesystem not dynamodb